src/context.jl

```diff
@@ -0,0 +1,44 @@
+##
+# Implements contextual dispatch through Cassette.jl
+# Goals:
+# - Rewrite common CPU functions to appropriate GPU intrinsics
+# 
+# TODO:
+# - error (erf, ...)
+# - pow
+# - min, max
+# - mod, rem
+# - gamma
+# - bessel
+# - distributions
+# - unsorted
+
+using Cassette
+
+Cassette.@context CUDACtx
+const cudactx = CUDACtx()
```

Comment:
unlikely to held much, but would `Cassette.disablehooks(CUDACtx())`help anything here (if you're not using hooks, then this at least reduces our "attack surface" for debugging stuff)